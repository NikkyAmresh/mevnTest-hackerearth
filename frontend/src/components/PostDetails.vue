<template>
  <div v-if="post">
    <h4>Post</h4>
    <div>
      <label><strong>Title:</strong></label> {{ post.title }}
    </div>
    <div>
      <label><strong>Description:</strong></label>
      {{ post.description }}
    </div>
    <div>
      <label><strong>Status:</strong></label>
      {{ post.published ? "Published" : "Pending" }}
    </div>
  </div>
</template>
<script>
export default {
  name: "post-details",
  props: {
    post: Object,
  },
};
</script>
