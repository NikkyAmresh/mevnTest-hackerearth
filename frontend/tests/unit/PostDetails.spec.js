import { shallowMount, RouterLinkStub } from "@vue/test-utils";
import PostDetails from "@/components/PostDetails.vue";

describe("PostDetails.vue", () => {
  it("renders props.post when passed for non-published posts", async () => {
    const post = { title: "xyz", description: "Abc", published: false };
    const wrapper = shallowMount(PostDetails, {
      propsData: {
        post,
      },
    });
    expect(wrapper.text()).toMatchSnapshot();
  });

  it("renders props.post when passed for published posts", async () => {
    const post = { title: "abc", description: "xyz", published: true };
    const wrapper = shallowMount(PostDetails, {
      props: {
        post,
      },
      stubs: {
        RouterLink: RouterLinkStub,
      },
    });
    expect(wrapper.text()).toMatchSnapshot();
  });
});
