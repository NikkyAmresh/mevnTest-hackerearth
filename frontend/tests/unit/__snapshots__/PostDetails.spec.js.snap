// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostDetails.vue renders props.post when passed for non-published posts 1`] = `"PostTitle: xyzDescription: AbcStatus: Pending"`;

exports[`PostDetails.vue renders props.post when passed for published posts 1`] = `"PostTitle: abcDescription: xyzStatus: Published"`;
